src/bindings/c/src/ov_core.cpp

```diff
@@ -132,6 +132,18 @@ ov_status_e ov_core_read_model_from_memory_buffer(const ov_core_t* core,
     return ov_status_e::OK;
 }
 
+ov_status_e ov_core_add_extension(const ov_core_t* core, const char* extension_path) {
+    if (!core || !extension_path) {
+        return ov_status_e::INVALID_C_PARAM;
+    }
+
+    try {
+        core->object->add_extension(extension_path);
+    }
+    CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
```

Comment:
This extension only works for CPU plugin, if we also need support custom ops extension, need add ov::intel_gpu::config_file property in C API's properties.

Reply:
I don't fully follow, can you please expand on this further or show some example/reference? I have primarily tested this with CPU.

Reply:
If we only need CPU plugin's extension for C APIs, this API is enough. While if also need GPU plugin's extension, we need export ov::intel_gpu::config_file property in C APIs as well.

Reply:
IIUC, one would load gpu extensions similar to what [this page](https://docs.openvino.ai/2024/documentation/openvino-extensibility/custom-gpu-operations.html) says, via `core.set_property`. I have added a commit which adds a property to `ov_property.h`, and also added tests. Since this should only be tested for GPU, I have made some additional changes to detect gpu/skip test similar to what [exists here](https://github.com/openvinotoolkit/openvino/blob/1e9768bd1600e99fb196cb76e11e9269eb0c0dae/src/bindings/c/tests/ov_remote_context_test.cpp#L25). Let me know if anything is missing.